### DISCUSSION TITLE
docs: Improvements in Decorators Documentation

### DISCUSSION THREAD

**Discussion by jefmenegazzo** [mikro-orm/mikro-orm#5940] [Ideas]
Upon reviewing the parameters of the decorators listed in [Decorators](https://mikro-orm.io/docs/decorators) and comparing them with the fields in the code (ManyToOneOptions, ManyToManyOptions, FormulaOptions, EntityOptions, etc.), I noticed that several parameters are not documented.

@Embedded: nothing about this decorator
Missing: entity, type, prefix, nullable, object, array, hidden, serializer, serializedName, groups, persist

[@Entity](https://mikro-orm.io/docs/decorators#entity)
Missing: virtual, expression

[@Property()](https://mikro-orm.io/docs/decorators#entity-properties)
Missing: name, fieldNames, precision, scale, autoincrement, onCreate, defaultRaw, formula, generated, ref, trackChanges, index, check, lazy, primary,  getter, setter,  serializedPrimaryKey, serializer, serializedName, extra, ignoreSchemaChanges

[@Index and @Unique](https://mikro-orm.io/docs/decorators#index-and-unique)
Missing: expression, options

[@Formula](https://mikro-orm.io/docs/decorators#formula)
It is missing the information that 'options' exist, and they are the same as the 'property' decorator.

[@ManyToOne](https://mikro-orm.io/docs/decorators#manytoone)
Missing: strategy, mapToPk, joinColumn, joinColumns, referenceColumnName, referencedColumnNames

[@OneToOne](https://mikro-orm.io/docs/decorators#onetoone)
Missing:  strategy, mapToPk, where, joinColumns, inverseJoinColumn, inverseJoinColumns, referenceColumnName

[@OneToMany](https://mikro-orm.io/docs/decorators#onetomany)
Missing:  where, joinColumns, inverseJoinColumns, referenceColumnName, entity, strategy

[ManyToMany](https://mikro-orm.io/docs/decorators#manytomany)
Missing:  pivotEntity, joinColumns, inverseJoinColumns, where, referenceColumnName, referencedColumnNames, deleteRule, updateRule, entity, cascade, eager, strategy

Given how some options inherit from others, I believe it's possible that a parameter might be used in one subclass/decorator but not in another. In this case, it would be helpful to list it in the documentation as N/A (not applicable), since the field is still accessible via code

**Comment by B4nan** [comment]
I guess it would be better to move the docs to jsdoc of the given symbol and only link to the [generated API docs](https://mikro-orm.io/api/core/interface/ManyToOneOptions) from that page, since that will be always up to date.

**Reply by B4nan** [reply]
The relation decorators now have all the jsdoc: 9e1818158feed9994ea1d0cd20c0d22e641e00e2
